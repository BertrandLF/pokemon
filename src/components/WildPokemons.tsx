import React from 'react';
import './styles/WildPokemons.css';
import PokeCell from './WildPokemon';
import Pokemon from './Pokemon';

interface WildPokemonsProps {
  pokemons: Map<number, Pokemon>;
  handleOnClick: (id: number) => void;
}
class WildPokemons extends React.Component<WildPokemonsProps> {

  cells() {
    return Array.from(this.props.pokemons.values())
            .sort((a, b) => a.id - b.id)
            .map((pokemon: Pokemon) =>
      <PokeCell
        key={pokemon.id}
        pokemon={pokemon}
        handleOnClick={this.props.handleOnClick}
      />
    );
  }

  render() {
    return (
      <section className="wild-pokemons">
        {this.cells()}
      </section>
    )
  }
}

export default WildPokemons;