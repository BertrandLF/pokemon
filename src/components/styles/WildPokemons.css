.wild-pokemons {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
  width: 50%;
  height: 100%;
  margin-right: 0.7em;
  background-color: rgb(145, 75, 177);
  border-top-left-radius: 1em;
  overflow: scroll;
}