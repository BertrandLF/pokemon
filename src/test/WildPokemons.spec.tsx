import React from 'react'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import WildPokemons from '../components/WildPokemons';
import { pokemonMap } from './PokemonFixture';

it('displays wildPokemons given list of pokemons', () => {

  const onClick = () => {};
  const component = shallow(
    <WildPokemons
      pokemons={pokemonMap}
      handleOnClick={onClick}>
    </WildPokemons>
  );

  expect(toJson(component)).toMatchSnapshot();
})

