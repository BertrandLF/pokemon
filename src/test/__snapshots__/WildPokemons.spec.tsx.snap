// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays wildPokemons given list of pokemons 1`] = `
<section
  className="wild-pokemons"
>
  <WildPokemon
    handleOnClick={[Function]}
    key="1"
    pokemon={
      Pokemon {
        "id": 1,
        "name": "balbusaur",
        "sprite": "https://someUrl/someOtherImage.png",
        "type": "poison",
      }
    }
  />
  <WildPokemon
    handleOnClick={[Function]}
    key="54"
    pokemon={
      Pokemon {
        "id": 54,
        "name": "psyduck",
        "sprite": "https://someUrl/someImage.png",
        "type": "water",
      }
    }
  />
</section>
`;
